javascript/spec/test.js

```diff
@@ -0,0 +1,66 @@
+import { expect } from 'chai'
+import oo from '../object_oriented.js'
+import functional from '../functional.js'
+import imperative from '../imperative.js'
+
+const test = (input, output) => {
+  const paradigms = [
+    ['Object Oriented', oo]
+    , ['Functional', functional]
+    , ['Imperative', imperative]
+  ]
+
+  for (let i in paradigms) {
+    const [name, fn] = paradigms[i]
+    it(name, () => {
+      expect(fn(input)).to.equal(output)
+    })
+  }
+}
+
+const nest = (element) => {
```

Comment:
I don't quite understand how this function works. Maybe its WIP, but in its current implementation it doesn't make sense to me. The value of `input` is being set to `test` and calling the `test` function. It means that the function `test` would always be called with the first argument as `test`.